src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6474,6 +6474,13 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
                                 Debug.Assert(!leftType.IsDynamic());
                                 Debug.Assert(IsPotentialColorColorReceiver(left, leftType));
 
+                                if (leftSymbol is SourceFieldSymbolWithSyntaxReference fieldLeft)
```

Comment:
>leftSymbol is SourceFieldSymbolWithSyntaxReference fieldLeft [](http://example.com/codeflow?start=36&length=60)

It is not obvious to me why only these symbols get special treatment.

Reply:
I guess that is because ```ConstantFieldsInProgress``` tracks only these symbols.

Reply:
The only symbols that encounter this bug are fields, and this is the only type that is being tracked by `ConstantFieldsInProgress` as you said. What kind of affected scenarios are you considering could be missed?